build_runner/lib/src/logging/std_io_logging.dart

```diff
@@ -57,3 +57,16 @@ void stdIOLogListener(LogRecord record, {bool verbose}) {
     stdout.write(message);
   }
 }
+
+/// Filter out the Logger names known to come from `build_runner`.
+String _loggerName(LogRecord record, bool verbose) {
+  var knownNames = const [
+    'Entrypoint',
+    'Build',
+    'BuildDefinition',
+    'Heartbeat'
+  ];
+  return verbose || !knownNames.contains(record.loggerName)
```

Comment:
looks like this is where we would only add the newline for >= WARNING?

Reply:
Yeah, I can add a level check here.